nixos/modules/system/boot/systemd/initrd.nix

```diff
@@ -359,6 +405,45 @@ in {
       "tpm-tis"
     ] ++ lib.optional (pkgs.stdenv.hostPlatform.system != "riscv64-linux") "tpm-crb";
 
+    boot.initrd.environment.etc = {
+      "systemd/system".source = stage1Units;
+
+      "systemd/system.conf".text = ''
+        [Manager]
+        DefaultEnvironment=PATH=/bin:/sbin
+        ManagerEnvironment=${lib.concatStringsSep " " (lib.mapAttrsToList (n: v: "${n}=${lib.escapeShellArg v}") cfg.managerEnvironment)}
+        ${cfg.extraConfig}
+      '';
+
+      "modules-load.d/nixos.conf".text = concatStringsSep "\n" config.boot.initrd.kernelModules;
+
+      "passwd".text = ''
+        ${lib.concatStringsSep "\n" (lib.mapAttrsToList (n: { uid, group }: let
+          g = cfg.groups.${group};
+        in "${n}:x:${toString uid}:${toString g.gid}:::") cfg.users)}
```

Comment:
The emergency shell complains `Could not chdir to home directory : No such file or directory, Logging in with home = "/"`
Set root account home to  '/root', and other accounts to '/var/empty'?